src/expr/src/linear.rs

```diff
@@ -625,7 +625,6 @@ impl MapFilterProject {
     /// instance, and so we should ensure that `self` is optimized to not reference
     /// columns that are not demanded.
     pub fn permute(&mut self, mut shuffle: HashMap<usize, usize>, new_input_arity: usize) {
-        self.remove_undemanded();
```

Comment:
`permute` seems to be used in a few other places, which may rely on this (idk). If so, perhaps we could pass a flag to indicate whether the internal layout is relied upon.

Reply:
I now support removing this call, in that `permute` is not tasked with re-optimization as a result of discarding output columns. We should consider the uses of `permute` and think about putting `self.remove_undemanded()` after each of those calls.

Reply:
Or, if we want `permute` to re-optimize (a legit take) we should not call `self.permute()` in a MFP with temporal filters.